internal/addrs/parse_target.go

```diff
@@ -51,8 +51,8 @@ func ParseTarget(traversal hcl.Traversal) (*Target, tfdiags.Diagnostics) {
 	}
 
 	var subject Targetable
-	switch {
-	case riAddr.Resource.Key == NoKey:
+	switch riAddr.Resource.Key {
+	case NoKey:
```

Comment:
I'm curious what lint rule this is addressing, just so I can know what new style rule I need to now follow related to switch statements. :grinning: 

With the benefit of hindsight, I think I'd probably write this fresh today in a different way:

```go
subject := Targetable(riAddr)
if subject.Resource.Key == NoKey {
    // We always assume yada yada yada
    subject = riAddr.ContainingResource()
}
```

Not intended as a blocker for merging, though. We can come back and improve this code further another day if we want to.